staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2328,6 +2328,9 @@ type ResourceRequirements struct {
 }
 
 // ResourceClaim references one entry in PodSpec.ResourceClaims.
+// This struct must be atomic as it is used within a slice with the
```

Comment:
Hmm, this line is going to be picked up in openapi, I wonder if I can hack this to prevent it being picked up there

In the past I've put `+` at the beginning of the line to prevent generators including dev targeted comments, do we think that would be appropriate here?

Reply:
I don't think this should be allowed tbh, though that looks like something can be fixed by having `listType=map`, `listMapKey=name`?

With regards to validating these, yeah, we should do a better job.

https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy

I don't know if the validation is correct, it should probably only allow scalar types.

Reply:
> I don't think this should be allowed tbh, though that looks like something can be fixed by having listType=map, listMapKey=name?

In a more complex example where you do want a set, eg with 3 or 4 members in the struct, would that then mean you have to make every field a map key to get the same semantic?

Reply:
I guess, but the keys have to be scalars too

Reply:
> I don't think this should be allowed tbh, though that looks like something can be fixed by having listType=map, listMapKey=name?

Would this be your suggested path forward for this particular API then? Convert it to a map?